modules/standard/ChapelEnv.chpl

```diff
@@ -45,6 +48,7 @@ module ChapelEnv {
   
   /* See :ref:`readme-chplenv.CHPL_HOME` for more information. */
   pragma "last resort"
+  pragma "no doc"
```

Comment:
It's not in the scope for this particular PR but it's kind of annoying that you have to manually flag everything "no doc" (or is there something in here that does get doc'd?).  Could we make that pragma work on the module scope?

Reply:
Ah, so is the suggestion to apply it to the module's block instead of the module itself?

Reply:
Seems like there are two use cases we'd care about:

1. no doc everything (the module and everything inside)
2. doc the module but no doc everything inside

Where 1 is the current behavior for putting "no doc" on a module.

There's also "no doc the module but doc everything inside it" but I don't think that would be useful so we can ignore that.

Anyway, would it be possible to apply a pragma at a scope level like this?

```
module ChapelEnv
pragma "no doc"  // Applies to '{' not module
{
```

Or is that too ugly\weird?  I suppose we could also make a new pragma "no doc contents".

Maybe use case 2 (doc the module but not the things inside it) is rare enough it's not worth thinking about, but if shows up in more than a couple places I'd be nice if we didn't have to "no doc" each individual field by hand.

Reply:
I don't know that we can apply pragmas like that today, I think that'd be a syntax change in addition to a change to `pragma "no doc"` itself?  `pragma "no doc contents"` would probably be the most straight-forward addition.  I suspect case 2 only comes up for deprecated modules, maybe this is a job for the deprecated keyword?

Reply:
I think `pragma "no doc contents"` would be good enough and easy to do. Regarding deprecated modules, I don't think it's necessarily the case that a deprecated module should not include documentation (of deprecated functions), but that's not something I have a strong opinion on.

Reply:
I'm trying to turn this thread into something actionable for this PR.  At first I thought there was "no action" and it was all just discussion about potential future improvements, but Michael's final comment may be suggesting that he'd prefer me to remove all the `pragma "no doc"`s?  I took my cue on this from the way the Spawn module shows up in 1.25, which Lydia reviewed and @daviditen implemented: https://chapel-lang.org/docs/modules/standard/Spawn.html   Checking, I'm also seeing that deprecated Regexp doesn't generate any documentation at all, and that if it did (were added back to the Makefile), it wouldn't either because it's empty.

So: Michael, how strongly do you feel about this, and Lydia and Andy, which parts of his comment were you thumbs-upping?

Reply:
I didn't interpret Michael's comment as implying a blanket statement (and thus a change for you), rather the opposite - that there were some situations where a deprecated module might want to continue to include documentation (but not necessarily saying anything about this particular case).

Reply:
So I was thumbs-upping that concept and `pragma "no doc contents"` being easy

Reply:
> Lydia and Andy, which parts of his comment were you thumbs-upping?

I'm also not sure if we'd want the `deprecated` keyword to imply anything about whether or not to document contents so I'm thumbing up that I'd be happy with a `pragma "no doc contents"` solution.

Reply:
Yes I think think the approach in this PR is good to go.